modules/seqkit/replace/main.nf

```diff
@@ -0,0 +1,41 @@
+process SEQKIT_REPLACE {
+    tag "$meta.id"
+    label 'process_low'
+
+    conda (params.enable_conda ? "bioconda::seqkit=2.1.0" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/seqkit:2.1.0--h9ee0642_0':
+        'quay.io/biocontainers/seqkit:2.1.0--h9ee0642_0' }"
+
+    input:
+    tuple val(meta), path(fastx)
+
+    output:
+    tuple val(meta), path("*.fast*"), emit: fastx
+    path "versions.yml"           , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def extension = "fastq"
+    if ("$fastx" ==~ /.+\.fasta|.+\.fasta.gz|.+\.fa|.+\.fa.gz|.+\.fas|.+\.fas.gz|.+\.fna|.+\.fna.gz/) {
```

Comment:
I wonder if just making this an input value `fastq` or `fasta` might be better to avoid all the potential fasta extensions

Reply:
It can take fasta or fastq input, and attempts to be semi-intelligent.

Reply:
I see your point on adding complexity, but I think the extra code doesn't hurt it in the long run, and adds decent compatibility with other modules outputs without too much additional configuration needed.